src/ucp/tag/eager_rcv.c

```diff
@@ -259,6 +259,8 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_eager_offload_sync_ack_handler,
     ucs_queue_for_each_safe(sreq, iter, queue, send.tag_offload.queue) {
         if ((sreq->send.tag_offload.ssend_tag == rep_hdr->sender_tag) &&
             (ucp_ep_local_id(sreq->send.ep) == rep_hdr->ep_id)) {
+            ucp_worker_del_request_id(worker, sreq,
+                                      sreq->send.tag_offload.sreq_id);
```

Comment:
is it really needed here - can replace by assert?

Reply:
replaced by assert